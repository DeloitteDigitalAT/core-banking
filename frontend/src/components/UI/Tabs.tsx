import React from "react";

interface ElementType {
  name: string;
  id: number;
  active: boolean;
}

interface TabsType {
  elements: ElementType[];
  setElements: (el: ElementType[]) => void;
}

const Tabs: React.FC<TabsType> = properties => {
  const { elements, setElements } = properties;
  const setActiveElement = (id: number) => {
    const updElements = elements.map(element => {
      if (element.id === id) {
        element.active = true;
      } else {
        element.active = false;
      }
      return element;
    });
    setElements(updElements);
  };
  return (
    <ul className="list-reset flex border-b mb-4">
      {elements.map((el, id) => (
        <li className="-mb-px mr-1" key={id}>
          <div
            className={`bg-white inline-block py-2 px-4 text-sm  cursor-pointer ${
              el.active
                ? "font-semibold text-gray-800 border-l border-t border-r rounded-t"
                : "text-gray-400 hover:text-gray-800"
            }`}
            onClick={() => setActiveElement(el.id)}
          >
            {el.name}
          </div>
        </li>
      ))}
    </ul>
  );
};

export default Tabs;
