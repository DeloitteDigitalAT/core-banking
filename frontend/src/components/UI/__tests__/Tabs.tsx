import React from "react";
import renderer from "react-test-renderer";
import Tabs from "../Tabs";

const mockElements = [
  {
    name: "tab name 1",
    id: 1,
    active: true,
  },
  {
    name: "tab name 2",
    id: 2,
    active: false,
  },
];

describe("UI/Tabs component", () => {
  test("snapshot renders", () => {
    const tree = renderer.create(<Tabs elements={mockElements} setElements={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
