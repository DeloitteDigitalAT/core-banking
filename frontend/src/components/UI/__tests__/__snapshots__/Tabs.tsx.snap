// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Tabs component snapshot renders 1`] = `
<ul
  className="list-reset flex border-b mb-4"
>
  <li
    className="-mb-px mr-1"
  >
    <div
      className="bg-white inline-block py-2 px-4 text-sm  cursor-pointer font-semibold text-gray-800 border-l border-t border-r rounded-t"
      onClick={[Function]}
    >
      tab name 1
    </div>
  </li>
  <li
    className="-mb-px mr-1"
  >
    <div
      className="bg-white inline-block py-2 px-4 text-sm  cursor-pointer text-gray-400 hover:text-gray-800"
      onClick={[Function]}
    >
      tab name 2
    </div>
  </li>
</ul>
`;
